app/src/full/java/io/homeassistant/companion/android/settings/SettingsWearViewModel.kt

```diff
@@ -0,0 +1,117 @@
+package io.homeassistant.companion.android.settings
+
+import android.app.Activity
+import android.net.Uri
+import android.util.Log
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateMapOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.android.gms.tasks.Tasks
+import com.google.android.gms.wearable.CapabilityClient
+import com.google.android.gms.wearable.DataItem
+import com.google.android.gms.wearable.DataMap
+import com.google.android.gms.wearable.DataMapItem
+import com.google.android.gms.wearable.PutDataMapRequest
+import com.google.android.gms.wearable.Wearable
+import io.homeassistant.companion.android.common.data.integration.Entity
+import io.homeassistant.companion.android.common.data.integration.IntegrationRepository
+import kotlinx.coroutines.launch
+
+class SettingsWearViewModel : ViewModel() {
+
+    private lateinit var integrationUseCase: IntegrationRepository
+
+    private val TAG = "SettingsWearViewModel"
+    private val CAPABILITY_WEAR_FAVORITES = "send_home_favorites"
+
+    var entities = mutableStateMapOf<String, Entity<*>>()
+        private set
+    var favoriteEntityIds = mutableStateListOf<String>()
+
+    fun init(integrationUseCase: IntegrationRepository) {
+        this.integrationUseCase = integrationUseCase
+        loadEntities()
+    }
+
+    private fun loadEntities() {
+        viewModelScope.launch {
+            integrationUseCase.getEntities().forEach {
+                entities[it.entityId] = it
+            }
+        }
+    }
+
+    fun saveHomeFavorites(data: String, item: DataItem) {
+        getFavorites(DataMapItem.fromDataItem(item).dataMap)
+        favoriteEntityIds.clear()
+        favoriteEntityIds.addAll(
+            data.removeSurrounding("[", "]").split(", ").map { it }
+        )
+    }
+
+    fun onEntitySelected(checked: Boolean, entityId: String, activity: Activity) {
+        if (checked)
+            favoriteEntityIds.add(entityId)
+        else
+            favoriteEntityIds.remove(entityId)
+        sendHomeFavorites(favoriteEntityIds.toList(), activity)
+    }
+
+    private fun sendHomeFavorites(favoritesList: List<String>, activity: Activity) = viewModelScope.launch {

```

Comment:
I think we need to change this to an `AndroidViewModel` rather than passing the activity:
> If the ViewModel needs the Application context, for example to find a system service, it can extend the AndroidViewModel class and have a constructor that receives the Application in the constructor, since Application class extends Context.

Reply:
done